pkg/reconciler/ingress/controller.go

```diff
@@ -129,15 +143,24 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 		r.kubeClient, endpointsInformer.Lister(), network.GetClusterDomainName(), endpointsTracker, logger)
 	r.ingressTranslator = &ingressTranslator
 
-	snapshot, err := r.caches.ToEnvoySnapshot()
-	if err != nil {
-		panic(err)
-	}
-
-	err = r.xdsServer.SetSnapshot(&snapshot, nodeID)
-	if err != nil {
-		panic(err)
-	}
+	// Let's start the management server when our cache is in sync, to avoid sending an incomplete configuration
+	// to an already running gateway container. If the cache is not warmed up after "cacheWarmUPTimeout" we just
+	// start the server as somehow we couldn't sync.
+	go func() {
+		waitForCache(logger, caches)
+
+		snapshot, err := r.caches.ToEnvoySnapshot()
+		if err != nil {
+			panic(err)
```

Comment:
Use `logger.Fatalw("Failed to create snapshot", zap.Error(err))`. Same suggestion below.